File: Data/MusicRecord.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MaestroNotes.Data
{
    public class MusicRecord
    {
        [Key]
        public int Id { get; set; } = 0;

        [MaxLength(200)]
        public string Bezeichnung { get; set; } = "";

        public DateTime Datum { get; set; } = DateTime.Now;
        [MaxLength(64)]
        public string Spielsaison { get; set; } = $"{DateTime.Now.Year}/{(DateTime.Now.Year+1)%100}";

        [MaxLength(2000)]
        public string Bewertung { get; set; } = "";

        // Relationships
        public int? DirigentId { get; set; }
        [ForeignKey("DirigentId")]
        public Dirigent? Dirigent { get; set; }

        public int? OrchesterId { get; set; }
        [ForeignKey("OrchesterId")]
        public Orchester? Orchester { get; set; }

        public int? OrtId { get; set; }
        [ForeignKey("OrtId")]
        public Ort? OrtEntity { get; set; }

        public List<Werk> Werke { get; set; } = new();

        public List<Solist> Solisten { get; set; } = new();


        public void Init(MusicRecord n)
        {
            Id = n.Id;
            Bezeichnung = n.Bezeichnung;

            Dirigent = n.Dirigent;
            DirigentId = n.DirigentId;

            Orchester = n.Orchester;
            OrchesterId = n.OrchesterId;

            OrtEntity = n.OrtEntity;
            OrtId = n.OrtId;

            Werke = n.Werke;
            Solisten = n.Solisten;

            Datum = n.Datum;
            Bewertung = n.Bewertung;
            Spielsaison = n.Spielsaison;
        }
    }
}

File: php-app/app/Models/MusicRecord.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class MusicRecord extends Model
{
    use HasFactory;

    protected $table = 'music_records';

    protected $fillable = [
        'bezeichnung',
        'datum',
        'spielsaison',
        'bewertung',
        'dirigent_id',
        'orchester_id',
        'ort_id',
    ];

    protected $casts = [
        'datum' => 'datetime',
    ];

    public function dirigent()
    {
        return $this->belongsTo(Dirigent::class);
    }

    public function orchester()
    {
        return $this->belongsTo(Orchester::class);
    }

    public function ort()
    {
        return $this->belongsTo(Ort::class);
    }

    public function werke()
    {
        return $this->belongsToMany(Werk::class, 'music_record_werk');
    }

    public function solisten()
    {
        return $this->belongsToMany(Solist::class, 'music_record_solist');
    }

    public function documents()
    {
        return $this->hasMany(Document::class);
    }
}
